ui/src/containers/Visualisations/components/FilterForm.js

```diff
@@ -69,6 +69,7 @@ const FilterForm = ({
       <div className="form-group">
         <ColorPicker
           color={filter.get('color', VISUALISATION_COLORS[index])}
+          onChangeHashToForceRender={filter.hashCode()}
```

Comment:
Essentially there seem to be some checks in one of the HOCs to prevent re-renders when non-function props aren't changed. Since the color will be the same after the query is changed, it meant that component was not re-rendered and therefore the onChange function still held a reference to the old filter which contained the old query. When the color was changed, this meant that the function updated the entire filter with the old query.